android/src/main/java/com/sensormanager/OrientationRecord.java

```diff
@@ -105,11 +105,15 @@ public void onSensorChanged(SensorEvent sensorEvent) {
             roll = 360 - (0 - roll);
           }
 
+          i++;
```

Comment:
I think it's better to move that righ under `long curTime = System.currentTimeMillis();`. Not to make excess work with geting orientation and performing data to send event.